azure-quantum/azure/quantum/optimization/problem.py

```diff
@@ -222,19 +330,36 @@ def to_blob(self, compress: bool = False) -> bytes:
         """Convert problem data to a binary blob.
 
         :param compress: Compress the blob using gzip, defaults to None
-        :type compress: bool, optional
+        :type compress: bool, optional 
         :return: Blob data
         :rtype: bytes
         """
-        problem_json = self.serialize()
-        logger.debug("Problem json: " + problem_json)
+        input_problem = self.serialize()
+        debug_input_string = input_problem if type(input_problem) is str else b''.join( input_problem).decode()
+        logger.debug("Input Problem: " + debug_input_string)
         data = io.BytesIO()
 
-        if compress:
+        if self.serialization_type == "application/x-protobuf":
```

Comment:
this should go into the existing `serialize_to_proto` as it only applies to protobuf